dev/io.openliberty.org.apache.myfaces.4.0_fat/fat/src/io/openliberty/org/apache/myfaces40/fat/tests/ClientWindowScopedTest.java

```diff
@@ -0,0 +1,169 @@
+import com.gargoylesoftware.htmlunit.WebClient;
+import com.gargoylesoftware.htmlunit.html.HtmlPage;
+import com.ibm.websphere.simplicity.RemoteFile;
+import com.ibm.websphere.simplicity.ShrinkHelper;
+
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.custom.junit.runner.Mode;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.HttpUtils;
+import io.openliberty.org.apache.myfaces40.fat.FATSuite;
+
+/**
+ * Test the new ClientWindowScoped context for Faces-4.0
+ * Full test because scanning logs is slow.
+ */
+@Mode(FULL)
+@RunWith(FATRunner.class)
+public class ClientWindowScopedTest {
+
+    private static final String APP_NAME = "ClientWindowScopedTest";
+
+    @Server("faces40_clientWindowScopedTest")
+    public static LibertyServer server;
+
+    @Rule
+    public TestName name = new TestName();
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        ShrinkHelper.defaultDropinApp(server, APP_NAME + ".war", "io.openliberty.org.apache.faces40.fat.client.window");
+
+        server.startServer(ClientWindowScopedTest.class.getSimpleName() + ".log");
+    }
+
+    @AfterClass
+    public static void testCleanup() throws Exception {
+        // Stop the server
+        if (server != null && server.isStarted()) {
+            server.stopServer();
+        }
+    }
+
+    /**
+     * Ensure the same client window is used when navigating between pages in a flow.
+     * Assertion: ClientWindowScoped superscedes FlowScoped
+     */
+    @Test
+    public void testClientWindowSupersedesFlow() throws Exception {
+        try (WebClient webClient = new WebClient()) {
+            URL url = HttpUtils.createURL(server, "/" + APP_NAME + "/clientWindowHierarchy.xhtml");
+            HtmlPage page = (HtmlPage) webClient.getPage(url);
+            FATSuite.logOutputForDebugging(server, page.asXml(), name.getMethodName() + ".html");
+
+            page = page.getHtmlElementById("form:enter").click();
+            String flow1Result = page.getHtmlElementById("result1").asText();
+            FATSuite.logOutputForDebugging(server, page.asXml(), name.getMethodName() + ".enter.html");
+
+            page = page.getHtmlElementById("form:next").click();
+            String flow2Result = page.getHtmlElementById("result2").asText();
+            FATSuite.logOutputForDebugging(server, page.asXml(), name.getMethodName() + ".next.html");
+
+            page = page.getHtmlElementById("form:exit").click();
+            FATSuite.logOutputForDebugging(server, page.asXml(), name.getMethodName() + ".exit.html");
+
+            assertEquals("Expected FlowScoped beans to be under the same ClientWindowScope", flow1Result, flow2Result);
+        }
+    }
+
+    /**
+     * Ensure a new client window is used when accessing a new view.
+     * Assertion: ViewScoped supersedes ClientWindowScoped
```

Comment:
The `Assertion: ViewScoped supersedes ClientWindowScoped` doesn't make sense to me?  What do you mean by ViewScoped? 

Maybe delete this comment?

Reply:
Ah, but you use the same description in the test names.